pkg/kds/v2/store/sync.go

```diff
@@ -86,8 +88,15 @@ func (s *syncResourceStore) Sync(syncCtx context.Context, upstreamResponse clien
 	if err != nil {
 		return err
 	}
-	if err := s.resourceStore.List(ctx, downstream); err != nil {
-		return err
+	if len(upstream.GetItems()) >= s.maxListQueryElements || upstreamResponse.IsInitialRequest {
```

Comment:
What is the reason for doing this type of limit? I'm wondering if this should be a parameter of the store instead of KDS.